File: kotlin/mth-core/src/main/kotlin/mth/core/MTTSFlags.kt
package mth.core

@JvmInline
value class MTTSFlags(val rawValue: Int = 0) {
    operator fun contains(flag: MTTSFlags): Boolean = rawValue and flag.rawValue == flag.rawValue

    companion object {
        val ANSI = MTTSFlags(1 shl 0)
        val VT100 = MTTSFlags(1 shl 1)
        val UTF8 = MTTSFlags(1 shl 2)
        val COLORS_256 = MTTSFlags(1 shl 3)
        val MOUSE_TRACKING = MTTSFlags(1 shl 4)
        val COLOR_PALETTE = MTTSFlags(1 shl 5)
        val SCREEN_READER = MTTSFlags(1 shl 6)
        val PROXY = MTTSFlags(1 shl 7)
        val TRUE_COLOR = MTTSFlags(1 shl 8)
    }
}

File: Sources/SwiftMTHCore/MTTSFlags.swift
/// MTTS (Mud Terminal Type Standard) capability flags.
///
/// Reported by the client via terminal type subnegotiation
/// in the format "MTTS <bitmask>".
public struct MTTSFlags: OptionSet, Sendable {
    public let rawValue: Int

    public init(rawValue: Int) {
        self.rawValue = rawValue
    }

    public static let ansi           = MTTSFlags(rawValue: 1 << 0)
    public static let vt100          = MTTSFlags(rawValue: 1 << 1)
    public static let utf8           = MTTSFlags(rawValue: 1 << 2)
    public static let colors256      = MTTSFlags(rawValue: 1 << 3)
    public static let mouseTracking  = MTTSFlags(rawValue: 1 << 4)
    public static let colorPalette   = MTTSFlags(rawValue: 1 << 5)
    public static let screenReader   = MTTSFlags(rawValue: 1 << 6)
    public static let proxy          = MTTSFlags(rawValue: 1 << 7)
    public static let trueColor      = MTTSFlags(rawValue: 1 << 8)
    public static let mnes           = MTTSFlags(rawValue: 1 << 9)
    public static let mslp           = MTTSFlags(rawValue: 1 << 10)
    public static let ssl            = MTTSFlags(rawValue: 1 << 11)
}
